### PR TITLE
torch benchmark policy gradient

### PR DESCRIPTION
**Pull request by zenghsh3**

### CODE REVIEW

File: parl/algorithms/torch/policy_gradient.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+        prob = self.model(obs)
+        sample_actions = Categorical(prob).sample()
+        return sample_actions
+
+    def learn(self, obs, action, reward):
+        """Update model with policy gradient algorithm
+
+        Args:
+            obs (torch.tensor): shape of (*, obs_dim)
```
Comment (yobobobo) [inline parl/algorithms/torch/policy_gradient.py]: replace * with `batch_size`.

File: parl/algorithms/torch/policy_gradient.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        """
+        assert isinstance(lr, float)
+
+        self.model = model
+        device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+        self.model.to(device)
+        self.optimizer = optim.Adam(self.model.parameters(), lr=lr)
+
+    def predict(self, obs):
+        """Predict actions of maximum probability
```
Comment (yobobobo) [inline parl/algorithms/torch/policy_gradient.py]: Predict the action with maximum probability

File: parl/algorithms/torch/policy_gradient.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+        Args:
+            obs (torch.tensor): shape of (*, obs_dim)
```
Comment (yobobobo) [inline parl/algorithms/torch/policy_gradient.py]: replace `*` with `batch_size`.

File: parl/algorithms/torch/policy_gradient.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+        Returns:
+            sample_actions (torch.tensor): shape of (*, 1)
+        """
+
+        prob = self.model(obs)
+        sample_actions = Categorical(prob).sample()
```
Comment (yobobobo) [inline parl/algorithms/torch/policy_gradient.py]: Could you use the original torch APIs like `softmax` to implement this function? I'm afraid that beginners are not familiar with the API `Categorical` used here.

File: parl/algorithms/torch/policy_gradient.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+            action (torch.tensor): shape of (*, 1)
+            reward (torch.tensor): shape of (*, 1)
+
+        Returns:
+            loss (torch.tensor): shape of (1)
+
+        """
+        prob = self.model(obs)
+
+        log_prob = Categorical(prob).log_prob(action)
```
Comment (yobobobo) [inline parl/algorithms/torch/policy_gradient.py]: Could you use the original torch APIs like `softmax` to implement this function? I'm afraid that beginners are not familiar with the API `Categorical` used here.

File: benchmark/torch/QuickStart/README.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
+## PyTorch benchmark Quick Start
+Train an agent with PARL(PyTorch benchmark) to solve the CartPole problem, a classical benchmark in RL.
```
Comment (yobobobo) [inline benchmark/torch/QuickStart/README.md]: remove `(PyTorch benchmark)`

File: parl/algorithms/torch/policy_gradient.py (hunk 2 above)
Comment (zenghsh3) [reply parl/algorithms/torch/policy_gradient.py]: done

File: parl/algorithms/torch/policy_gradient.py (hunk 3 above)
Comment (zenghsh3) [reply parl/algorithms/torch/policy_gradient.py]: But in the example, the shape of obs in predict/sample is (obs_dim). we don't need to expand dims to the shape of (1, obs_dim) in the torch.

File: benchmark/torch/QuickStart/README.md (hunk 6 above)
Comment (zenghsh3) [reply benchmark/torch/QuickStart/README.md]: done

File: parl/algorithms/torch/policy_gradient.py (hunk 5 above)
Comment (zenghsh3) [reply parl/algorithms/torch/policy_gradient.py:65]: ok, but I think using `Categorical` is the best practice.

Comment (TomorrowIsAnOtherDay) [reply parl/algorithms/torch/policy_gradient.py:65]: Yeah, this API is ok for the  implementation of more complex algorithms such as A2C, but we would like to use some commonly used APIs here for the beginners, avoiding that they have to understand what is `Categorical` here.

File: parl/algorithms/torch/policy_gradient.py (hunk 3 above)
Comment (TomorrowIsAnOtherDay) [reply parl/algorithms/torch/policy_gradient.py]: Alright, but `*` normally does not represent that it can be ignored here. Please update the comment as `(batch_size, obs_dim)` or `(obs_sim,)`.

File: parl/algorithms/torch/policy_gradient.py (hunk 1 above)
Comment (zenghsh3) [reply parl/algorithms/torch/policy_gradient.py]: done

File: parl/algorithms/torch/policy_gradient.py (hunk 3 above)
Comment (zenghsh3) [reply parl/algorithms/torch/policy_gradient.py]: ok

File: parl/algorithms/torch/policy_gradient.py (hunk 4 above)
Comment (zenghsh3) [reply parl/algorithms/torch/policy_gradient.py]: done

File: parl/algorithms/torch/policy_gradient.py (hunk 5 above)
Comment (zenghsh3) [reply parl/algorithms/torch/policy_gradient.py:65]: ok

File: parl/algorithms/torch/policy_gradient.py [hunk 7]
```diff
@@ -37,39 +37,24 @@ def __init__(self, model, lr):
         self.optimizer = optim.Adam(self.model.parameters(), lr=lr)
 
     def predict(self, obs):
-        """Predict actions of maximum probability
+        """Predict the probability of actions
```
Comment (TomorrowIsAnOtherDay) [inline parl/algorithms/torch/policy_gradient.py]: Predict the action with maximum probability.

**Review by TomorrowIsAnOtherDay — APPROVED** [review]
Looks good to me.
